src/crypto/set_hasher.rs

```diff
@@ -0,0 +1,100 @@
+use super::blake3_sequence::Blake3SeqNo;
+use crate::prelude::*;
+use crate::stable_hash::UnorderedAggregator;
+use blake3::Hasher;
+use lazy_static::lazy_static;
+use num_bigint::BigUint;
+use num_traits::identities::One;
+use std::default::Default;
+
+lazy_static! {
+    static ref P: BigUint = "50763434429823703141085322590076158163032399096130816327134180611270739679038131809123861970975131471260684737408234060876742190838745219274061025048845231234136148410311444604554192918702297959809128216170781389312847013812749872750274650041183009144583521632294518996531883338553737214586176414455965584933129379474747808392433032576309945590584603359054260866543918929486383805924215982747035136255123252119828736134723149397165643360162699752374292974151421555939481822911026769138419707577501643119472226283015793622652706604535623136902831581637275314074553942039263472515423713366344495524733341031029964603383".parse().unwrap();
+}
+
+/// Based on https://crypto.stackexchange.com/a/54546
+///
+/// The idea here is to use the SequenceNumber to unambiguously identify each
+/// field as within it's own database cell, and use an online order-independent
+/// aggregator of the cells to produce a final result.
+///
+/// Within this framework a huge struct can be hashed incrementally or even in
+/// parallel as long as sequence numbers are deterministically produced to
+/// identify parts within the struct. Conveniently, the SequenceNumber::skip
+/// method can be used to jump to parts of a vec or struct efficiently.
+pub struct SetHasher {
+    // TODO: (Performance). We want an int 2056 + 2048 = 4104 bit int.
```

Comment:
Are you saying that you plan to implement a crate / traits that can easily give 4104 bit ints, and once this is done, the set hash implementation should be more performant?

Reply:
Yes. The current choice `num-bigint::BigUint` requires multiple variable-length heap allocations for every mixin. It is possible to have a large int be allocated on the stack for a quick and easy performance gain.